### PR TITLE
Fix vm::range_lock, improve vm::check_addr

### LINKED ISSUE #9237: Regression: GT6 freeze on intro after #8909
After PR #8909, Gran Turismo 6 freezes on the SCE Presents intro part. TSX is unaffected.
image
[RPCS3.zip](https://github.com/RPCS3/rpcs3/files/5518932/RPCS3.zip)
Last working build is rpcs3-v0.0.13-11186-a28ab0a4

Ryzen 7 2700
RX 580 20.10.1
Windows 20H2

### PR DESCRIPTION
**Pull request by elad335**
Why do I have to suffer from all of those rewrites :think-hang:
I dont have even the power to even explain how the code works.

Fixes #9237

### CODE REVIEW

**Comment by elad335** [comment]
Confirmed works.

**Comment by Nekotekina** [comment]
One game shouldn't really get special codepaths. At least inline version need to get some optimization. Page info array can get disclosed at this point for direct access. Thus, if Size is specified, only one page needs check. Internal function is fine.

**Comment by elad335** [comment]
Fixed.

File: rpcs3/Emu/Memory/vm_locking.h [hunk 1]
```diff
@@ -52,6 +52,14 @@ namespace vm
 		// Old-style conditional constexpr
 		const u32 size = Size ? Size : _size;
 
+		// Assume address is aligned to size if less/equal than 16-bytes
+		if (Size - 1 <= 15u ? !vm::check_addr(begin) : !vm::check_addr(begin, vm::page_readable, size))
```
Comment (Nekotekina) [inline rpcs3/Emu/Memory/vm_locking.h:56]: Why max 16? I think it can be 4096. Plus you don't have to optimize constexpr checks.

**Review by Nekotekina — CHANGES_REQUESTED** [review]

**Comment by elad335** [comment]
Done.

**Pull request merged**